core/src/main/java/org/elasticsearch/common/io/Streams.java

```diff
@@ -241,4 +243,11 @@ public static void readAllLines(InputStream input, Callback<String> callback) th
             }
         }
     }
+
+    public static byte[] toByteArray(InputStream in) throws IOException {
+        try (BytesStreamOutput out = new BytesStreamOutput()) {
+            Streams.copy(in, out);
```

Comment:
`o.e.i.Streams.copy` will close both `in` and `out` after the copy is complete, but `ByteStreams.toByteArray` did not do this. This could potentially be problematic in the case of `in` here; we will need to think through this carefully.